File: backend/src/main/java/at/campus/backend/security/Roles.java
package at.campus.backend.security;

/**
 * Canonical role names used inside the backend.
 *
 * Incoming role headers are normalized to this uppercase format.
 */
public final class Roles {

    public static final String APPLICANT = "APPLICANT";
    public static final String STUDENT = "STUDENT";
    public static final String MODERATOR = "MODERATOR";

    private Roles() {
    }
}

File: frontend/src/shared/utils/roles.ts
export type NormalizedUserRole = 'APPLICANT' | 'STUDENT' | 'MODERATOR'

export function normalizeUserRole(role?: string | null): NormalizedUserRole | null {
  if (!role) {
    return null
  }

  const normalized = role.trim().toUpperCase()
  if (normalized === 'APPLICANT' || normalized === 'STUDENT' || normalized === 'MODERATOR') {
    return normalized
  }

  return null
}

export function isModeratorRole(role?: string | null): boolean {
  return normalizeUserRole(role) === 'MODERATOR'
}

export function isApplicantRole(role?: string | null): boolean {
  return normalizeUserRole(role) === 'APPLICANT'
}
